fil_fungible_token/src/token/mod.rs

```diff
@@ -294,6 +296,28 @@ where
         Ok(())
     }
 
+    /// Sets the allowance to a specified amount, returning the old allowance
+    pub fn set_allowance(
+        &mut self,
+        owner: &Address,
+        operator: &Address,
+        amount: &TokenAmount,
+    ) -> Result<TokenAmount> {
```

Comment:
This should `validate_allowance` (because it makes sense, and also consistency with `set_balance`).